master/internal/rm/kubernetesrm/pod.go

```diff
@@ -148,8 +158,20 @@ func (p *pod) Receive(ctx *actor.Context) error {
 	switch msg := ctx.Message().(type) {
 	case actor.PreStart:
 		ctx.AddLabels(p.logCtx)
-		if err := p.createPodSpecAndSubmit(ctx); err != nil {
-			return err
+		if p.restore {
+			if p.container.State == cproto.Running && !p.testLogStreamer {
```

Comment:
what is `testLogStreamer` even doing. it looks like it just reads the logs away in a loop.

Reply:
```testLogStreamer``` I believe is for mocking testing purposes to avoid creating the ```podLogStremter``` in unit tests? 

It seems ```podLogStreamer``` eventually sends logs as events to the allocation actor. I'm not entirely sure why we have ```podLogStreamer```  does when we have fluent?